src/components/stepper/step/step.ts

```diff
@@ -0,0 +1,133 @@
+import type { SbbStepLabelElement } from '../step-label';
+import type { SbbStepperElement } from '../stepper';
+
+import style from './step.scss?lit&inline';
+
+let nextId = 0;
+
+type ValidateEventDetails = {
+  currentIndex?: number;
+  currentStep?: SbbStepElement;
+  nextIndex?: number;
+  nextStep?: SbbStepElement;
+};
+
+/**
+ * Combined with a `sbb-stepper`, it displays a step's content.
+ *
+ * @slot - Use the unnamed slot to provide a content.
+ * @event {CustomEvent<ValidateEventDetails>} validate - Emits whenever step switch is triggered. Can be canceled.
+ */
+@customElement('sbb-step')
+@hostAttributes({
+  slot: 'step',
+  role: 'tabpanel',
+})
+export class SbbStepElement extends LitElement {
+  public static override styles: CSSResultGroup = style;
+  public static readonly events = {
+    validate: 'validate',
+  } as const;
+
+  /**
+   * Emits whenever step switch is triggered.
+   */
+  private _validate: EventEmitter<ValidateEventDetails> = new EventEmitter(
+    this,
+    SbbStepElement.events.validate,
+  );
+
+  private _abort = new ConnectedAbortController(this);
+  private _stepper: SbbStepperElement | null = null;
+  private _label: SbbStepLabelElement | null = null;
+  private _stepResizeObserver = new AgnosticResizeObserver((entries) =>
+    this._onStepElementResize(entries),
+  );
+
+  /**
+   * The label of the step.
+   */
+  public get label(): SbbStepLabelElement | null {
+    return this._label;
+  }
+
+  public validate(eventData: ValidateEventDetails): boolean {
+    return !!this._validate.emit(eventData);
+  }
+
+  private _handleClick(event: Event): void {
+    const composedPathElements = event
+      .composedPath()
+      .filter((el) => el instanceof window.HTMLElement);
+    if (composedPathElements.some((el) => this._isGoNextElement(el as HTMLElement))) {
+      this._stepper?.next();
+    } else if (composedPathElements.some((el) => this._isGoPreviousElement(el as HTMLElement))) {
+      this._stepper?.previous();
+    }
+  }
+
+  private _isGoNextElement(element: HTMLElement): boolean {
+    return element.hasAttribute('sbb-stepper-next') && !isValidAttribute(element, 'disabled');
+  }
+
+  private _isGoPreviousElement(element: HTMLElement): boolean {
+    return element.hasAttribute('sbb-stepper-previous') && !isValidAttribute(element, 'disabled');
+  }
+
+  private _onStepElementResize(entries: ResizeObserverEntry[]): void {
+    if (!this.hasAttribute('data-selected')) {
+      return;
+    }
+    for (const entry of entries) {
+      const contentHeight = Math.floor(entry.contentRect.height);
+      this._stepper?.style?.setProperty('--sbb-stepper-content-height', `${contentHeight}px`);
+    }
+  }
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    const signal = this._abort.signal;
+    this.id = this.id || `sbb-step-${nextId++}`;
+    this.addEventListener('click', (e) => this._handleClick(e), { signal });
+    this._stepper = this.closest('sbb-stepper');
```

Comment:
Same comment on `closest` as in step-label.ts

Reply:
Idk if I can use the optional chaining like that, anyway the `closest()` method should be null safe.